crates/test_utils/wasm/wasm_workspace/must_get/src/integrity.rs

```diff
@@ -1,11 +1,68 @@
 use hdi::prelude::*;
+use hdk::prelude::ChainFilter;
 
 #[hdk_entry_helper]
 #[derive(Clone)]
 pub struct Something(#[serde(with = "serde_bytes")] pub Vec<u8>);
 
+#[hdk_entry_helper]
+pub struct AgentsChain(pub AgentPubKey, pub ChainFilter);
+
+#[hdk_entry_helper]
+pub struct AgentsChainRec(pub AgentPubKey, pub ActionHash);
+
 #[hdk_entry_defs]
 #[unit_enum(EntryTypesUnit)]
 pub enum EntryTypes {
     Something(Something),
+    AgentsChain(AgentsChain),
+    AgentsChainRec(AgentsChainRec),
+}
+
+#[hdk_extern]
+fn validate(op: Op) -> ExternResult<ValidateCallbackResult> {
+    match op.to_type::<_, ()>()? {
+        OpType::StoreEntry(e) => match e {
+            OpEntry::CreateEntry {
+                entry_type: EntryTypes::AgentsChain(AgentsChain(author, filter)),
+                ..
+            } => {
+                must_get_agent_activity(author, filter)?;
+                return Ok(ValidateCallbackResult::Valid);
+            }
+            OpEntry::CreateEntry {
+                entry_type: EntryTypes::AgentsChainRec(AgentsChainRec(author, chain_top)),
+                ..
+            } => {
+                let mut filter = ChainFilter::new(chain_top).take(2);
+                loop {
+                    let chain = must_get_agent_activity(author.clone(), filter.clone())?;
+                    if chain.len() > 2 {
+                        return Ok(ValidateCallbackResult::Invalid(
+                            "Filter returned greater than 2".to_string(),
+                        ));
+                    }
+                    match chain.last() {
+                        Some(op) => {
+                            if op.action.action().action_seq() == 0 {
+                                return Ok(ValidateCallbackResult::Valid);
+                            } else {
+                                filter =
+                                    ChainFilter::new(op.action.action_address().clone()).take(2);
```

Comment:
You're filtering the chain 2 actions at a time until reaching the bottom? Seems like a good test, but I'm just curious why you're doing it this way.

Reply:
This is to test that our `must_get_agent_activity` recursive logic works. When this runs the first time it will actually fail with each recursion and go to the network then run the wasm again